pgcontents/tests/test_synchronization.py

```diff
@@ -425,3 +436,7 @@ def check_call(kwargs, expect_checkpoints):
         # `min_dt` and `max_dt` together isolate `middle_checkpoints`
         check_call({'min_dt': middle_min_dt, 'max_dt': end_min_dt},
                    middle_checkpoints)
+
+        # Dispose of all engines created during this test to prevent leaked
+        # database connections.
+        self.cleanup_pgcontents_managers(six.itervalues(managers))
```

Comment:
i would probably move this to an `addCleanup` call as soon as you've created the managers. If this test fails partway through, you'll still leak these.